tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_delete_thread.py

```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import SkipTest
+from marionette_driver import Wait
+from gaiatest import GaiaTestCase
+from gaiatest.apps.messages.app import Messages
+from gaiatest.apps.system.app import System
+from gaiatest.utils.plivo.plivo_util import PlivoUtil
+
+
+class TestSms(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+
+        try:
+            self.testvars['plivo']
+        except KeyError:
+            raise SkipTest('Plivo account details not present in test variables')
+
+        plivo = PlivoUtil(
+            self.testvars['plivo']['auth_id'],
+            self.testvars['plivo']['auth_token'],
+            self.testvars['plivo']['phone_number']
+        )
+
+        plivo.send_sms(to_number=self.environment.phone_numbers[0].replace('+', ''),
```

Comment:
I'd rather stop from using the data layer to send SMS. This makes us dependent from the internal APIs. Hence, we can use the data layer in an integration test, but not in an end to end test.

On another note, if we send an SMS to itself, we'd have only 1 thread to delete, instead of 2.